worker/caasoperator/state.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasoperator
+
+import "gopkg.in/juju/charm.v6"
+
+// LocalState is a cache of the state of the operator
+// It is generally compared to the remote state of the
+// the application as stored in the controller.
+type LocalState struct {
```

Comment:
can we call the file localstate.go?

Reply:
i didn't think we had enough "state" in juju so i added one more :-D